server/historian/packages/historian-base/src/routes/utils.ts

```diff
@@ -60,31 +60,48 @@ export function handleResponse<T>(
 	);
 }
 
-export async function createGitService(
-	config: nconf.Provider,
-	tenantId: string,
-	authorization: string,
-	tenantService: ITenantService,
-	cache?: ICache,
-	asyncLocalStorage?: AsyncLocalStorage<string>,
-	allowDisabledTenant = false,
-): Promise<RestGitService> {
+export class createGitServiceArgs {
+	config: nconf.Provider;
+	tenantId: string;
+	authorization: string;
+	tenantService: ITenantService;
+	cache?: ICache;
+	asyncLocalStorage?: AsyncLocalStorage<string>;
+	initialUpload?: boolean = false;
+	initialStorageName?: string;
+	allowDisabledTenant?: boolean = false;
+}
+
+export async function createGitService(createArgs: createGitServiceArgs): Promise<RestGitService> {
+	const {
+		config,
+		tenantId,
+		authorization,
+		tenantService,
+		cache,
+		asyncLocalStorage,
+		initialUpload,
+		initialStorageName,
```

Comment:
Seems like _initial_ is not necessary. _initialStorageName_  alone can do the job.

Reply:
I have thought about it, See this comments: https://github.com/microsoft/FluidFramework/pull/15264#discussion_r1177272689

There is a possibility that routerlicious and other component provide storage to historian, in that case, the storageName is not subject to initial upload anymore.